aws/aws-sdk-java-v2
# Add flexible checksum support for non-streaming cases

## Motivation and Context
- Flexible checksum support is needed for a few specific cases where a custom checksum can be added to a request
- This PR addresses the non-streaming cases for unsigned and signed payload
- A follow-up PR will be sent to address the streaming (aws-chunked) case, but there is no plan to support flexible checksums for event-streaming case or non-streaming CRT-V4a

## Modifications
- Copies `SdkChecksum` and related code from `sdk-core` to use internally without importing `sdk-core`
- Add new internal classes to handle flexible checksum cases, and remove the old digest-related stream and subscriber code
- Cleans up a couple of APIs that should be internal, not protected

## Testing
- Run new and existing tests, make sure no failures
- Run subscriber TCK test, make sure the implementation is compliant

## Types of changes


## Checklist



## License
